wasm/store.go

```diff
@@ -304,16 +305,18 @@ func (s *Store) Instantiate(module *Module, name string) error {
 	rollbackFuncs = nil
 
 	// Execute the start function.
+	ctx := context.Background()
 	if startIndex := module.StartSection; startIndex != nil {
 		f := instance.Functions[*startIndex]
-		if _, err := s.engine.Call(f); err != nil {
+		if _, err := s.engine.Call(ctx, f); err != nil {
 			return fmt.Errorf("calling start function failed: %v", err)
 		}
 	}
 	return nil
 }
 
-func (s *Store) CallFunction(moduleName, funcName string, params ...uint64) (results []uint64, resultTypes []ValueType, err error) {
+// Note: this API is unstable. See tetratelabs/wazero#170
+func (s *Store) CallFunction(ctx context.Context, moduleName, funcName string, params ...uint64) (results []uint64, resultTypes []ValueType, err error) {
```

Comment:
Should we fail early, fail late or default any nil context here?

Reply:
My gut says to set it to context.Background() so that host calls can assume a valid context is passed. That or fail/return an error.

Reply:
ok let's go with context.Background, and update the docs to say that's what it defaults if nil. TestStore_CallFunction should verify this is never nil